daemon/src/db/cull_old_dlcs.rs

```diff
@@ -1,127 +1,82 @@
-                _ => None,
-            };
-
-            event.map(|event| (id, event))
-        }))
-}
+    /// Cull old cfd rollover events which are not needed anymore
+    ///
+    /// We set the `dlc` field of events `RolloverCompleted` to null where the event is not the most
+    /// recent `RolloverCompleted` event.
+    async fn cull_rollover_completed(conn: &mut PoolConnection<Sqlite>) -> Result<()> {
+        let effected_rows = sqlx::query!(
+            r#"
+            update EVENTS
+                set
+                    data = (
+                        select json_set(events.data, '$.dlc', null)
+                        from events as inner_events
+                        where inner_events.id = EVENTS.id 
+                    )
+                where name = 'RolloverCompleted'
+                  and id not in (
+                    select max(id) from EVENTS as inner_events
+                    where inner_events.cfd_id = EVENTS.cfd_id
+                      and inner_events.name = EVENTS.name
+                );
+            "#
+        )
+        .execute(&mut *conn)
+        .await?
+        .rows_affected();
+
+        tracing::debug!(%effected_rows,"Culled RolloverCompleted events");
 
-/// Overwrite the contents of the `data` column for a row
-/// corresponding to `id` with the JSON blob originated from
-/// `updated_event`.
-async fn update_event_data(
-    conn: &mut PoolConnection<Sqlite>,
-    id: i64,
-    updated_event: EventKind,
-) -> Result<()> {
-    let (_, updated_data) = updated_event.to_json();
-
-    let query_result = sqlx::query!(
-        r#"
-        UPDATE
-            events
-        SET
-            data = $1
-        WHERE
-            id = $2
-        "#,
-        updated_data,
-        id,
-    )
-    .execute(&mut *conn)
-    .await?;
-
-    if query_result.rows_affected() != 1 {
-        anyhow::bail!("Failed to update event data");
+        Ok(())
     }
 
-    Ok(())
+    /// Cull `ContractSetupCompleted` events which are not needed anymore
+    ///
+    /// `ContractSetupCompleted` events are not needed anymore if there was at least one
+    /// `RolloverCompletedEvent` afterwards
+    async fn cull_contract_setup_completed(conn: &mut PoolConnection<Sqlite>) -> Result<()> {
+        let effected_rows = sqlx::query!(
+            r#"
+            update EVENTS
+                set
+                    data = (
+                        select json_set(events.data, '$.dlc', null)
+                        from events as inner_events
+                        where EVENTS.id = inner_events.id
+                    )
+                where name = 'ContractSetupCompleted'
+                  and cfd_id in (
+                    select distinct cfd_id from EVENTS as inner_events
+                        where inner_events.cfd_id = EVENTS.cfd_id
+                        and inner_events.name = 'RolloverCompleted'
+                    );
+            "#
+        )
```

Comment:
🔧 We normally use these constants for this.
```suggestion
                where name = $1
                  and cfd_id in (
                    select distinct cfd_id from EVENTS as inner_events
                        where inner_events.cfd_id = EVENTS.cfd_id
                        and inner_events.name = 'RolloverCompleted'
                    );
            "#,
            model::EventKind::CONTRACT_SETUP_COMPLETED_EVENT,
        )
```